### PR TITLE
[#4398] Support to get running taskInfo from worker

### PR DESCRIPTION
**Pull request by shenh062326**

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/server/TaskResource.java [hunk 1]
```diff
@@ -110,9 +111,17 @@ public TaskResource(
     @ResourceSecurity(INTERNAL_ONLY)
     @GET
     @Produces(MediaType.APPLICATION_JSON)
-    public List<TaskInfo> getAllTaskInfo(@Context UriInfo uriInfo)
+    public List<TaskInfo> getAllTaskInfo(@Context UriInfo uriInfo, @QueryParam("state") String stateFilter)
```
Comment (losipiuk) [inline presto-main/src/main/java/io/prestosql/server/TaskResource.java:114]: Adding possibility to filter out specifically on state field feels very arbitrary to me. Why not do the filtering on the caller side according to specific needs?

Comment (shenh062326) [reply presto-main/src/main/java/io/prestosql/server/TaskResource.java:114]: > Adding possibility to filter out specifically on state field feels very arbitrary to me. Why not do the filtering on the caller side according to specific needs?

We actually want to get running taskinfo just like get running queryInfo in QueryResource.java, but if there is no filtering state, we will get a lot of finished taskinfo.

Comment (losipiuk) [reply presto-main/src/main/java/io/prestosql/server/TaskResource.java:114]: I understand. But I do no necessarily see why is that a problem. Should be easy to filter out those even in shell using `jq`. And in proper scripting language even easier.

Comment (shenh062326) [reply presto-main/src/main/java/io/prestosql/server/TaskResource.java:114]: Yes, this is not a bug, it is an improvement. With this feature, my upgrade script does not need to download dozens of workers every few seconds, each worker 40-50MB taskinfo when checking whether the worker still has a running task.

Comment (losipiuk) [reply presto-main/src/main/java/io/prestosql/server/TaskResource.java:114]: Thanks for explaining motivation @shenh062326. I am leaning towards merging this one. @electrum can you chime in and shout if you feel that such an extension to protocol something you are very much against?

Comment (findepi) [inline presto-main/src/main/java/io/prestosql/server/TaskResource.java:114]: nit: you probably can `String stateFilter` -> `@Nullable TaskState stateFilter`  or even `Optional<TaskState> stateFilter`

**Comment by electrum** [comment]
Can you explain the high level goal here? This is an internal-only API (note the `@ResourceSecurity(INTERNAL_ONLY)` annotation). We should instead add a supported management API that can allow your use case.

**Comment by shenh062326** [comment]
> Can you explain the high level goal here? This is an internal-only API (note the `@ResourceSecurity(INTERNAL_ONLY)` annotation). We should instead add a supported management API that can allow your use case.

Since the worker can upgrade gracefully, we add a upgrade script,  which sent shutdown to workers, then wait for workers shutdown.  Now it need to download dozens of workers every few seconds , each worker 40-50MB taskInfos when checking whether the worker still has a running task. It will be better if we can only get running taskInfos.

Our internal presto version does not has the INTERNAL_ONLY annotation. If the INTERNAL_ONLY annotation cannot be accessed, how to add a management API?

**Comment by colebow** [comment]
:wave: @shenh062326 - this PR has become inactive. If you're still interested in working on it, please let us know, and we can try to get reviewers to help with that.

We're working on closing out old and inactive PRs, so if you're too busy or this has too many merge conflicts to be worth picking back up, we'll be making another pass to close it out in a few weeks.

**Pull request closed without merging**